Technical review: Add documentation for the geolocation element (#42772)
mdn/content: files/en-us/web/api/htmlgeolocationelement/index.md

---
title: HTMLGeolocationElement
slug: Web/API/HTMLGeolocationElement
page-type: web-api-interface
browser-compat: api.HTMLGeolocationElement
---

{{APIRef("HTML DOM")}}

> The **`HTMLGeolocationElement`** interface represents the {{htmlelement("geolocation")}} element, and provides access to its properties and events.

Reviewer: The accepted pattern is 

```suggestion
The **`HTMLGeolocationElement`** interface of the [Navigation API](/en-US/docs/Web/API/Navigation_API) represents the {{htmlelement("geolocation")}} element, and provides access to its properties and events.
```

  - : A boolean value indicating whether the `<geolocation>` element is valid or invalid (blocked).
- {{domxref("HTMLGeolocationElement.permissionStatus", "permissionStatus")}} {{readonlyinline}} {{experimental_inline}}
  - : A string representing the current permission status for the `geolocation` feature.
- {{domxref("HTMLGeolocationElement.position", "position")}} {{readonlyinline}} {{experimental_inline}}
  - : A {{domxref("GeolocationPosition")}} object representing the user's current position, in the event of successful location data retrieval.
- {{domxref("HTMLGeolocationElement.watch", "watch")}} {{experimental_inline}}
  - : A boolean value indicating whether the browser should request location data continuously, whenever the position of the user's device changes, or only once. Mirrors the value of the `<geolocation>` [`watch`](/en-US/docs/Web/HTML/Reference/Elements/geolocation#watch) attribute.

## Instance methods

_Inherits properties from its parent interface, {{domxref("HTMLElement")}}._

## Events

_Also inherits events from its parent interface, {{domxref("HTMLElement")}}._

- {{domxref("HTMLGeolocationElement.location_event", "location")}} {{experimental_inline}}

>   - : Fired whenever the browser receives location data, or error information when the location data request was unsuccessful.

Reviewer: I think
```suggestion
  - : Fired whenever the browser receives location data, or error information when a location data request was unsuccessful.
```

- {{domxref("HTMLGeolocationElement.promptaction_event", "promptaction")}} {{experimental_inline}}
  - : Fired whenever the user activates the `<geolocation>` element and selects an option from the resulting dialog, either to grant or deny `geolocation` permission.
- {{domxref("HTMLGeolocationElement.promptdismiss_event", "promptdismiss")}} {{experimental_inline}}
  - : Fired whenever the user activates the `<geolocation>` element and dismisses the resulting dialog, by pressing the "close" button or the <kbd>Esc</kbd> key.
- {{domxref("HTMLGeolocationElement.validationstatuschange_event", "validationstatuschange")}} {{experimental_inline}}
  - : Fired whenever the `<geolocation>` element's {{domxref("HTMLGeolocationElement.isValid", "isValid")}} value changes.

## Description

> The `HTMLGeolocationElement` interface represents the {{htmlelement("geolocation")}} element, which creates an interactive control to allow the user to share their location data with the page in a more consistent and intuitive way than rolling your own solution via the [Geolocation API](/en-US/docs/Web/API/Geolocation_API).

Reviewer: Yeah, I still find positioning an API purely by its historical context a bit odd. Maybe
```suggestion
The `HTMLGeolocationElement` interface represents the {{htmlelement("geolocation")}} element, which creates an interactive control to allow the user to share their location data with the page.
```

And you could mention " in a more consistent and intuitive way than rolling your own solution via the [Geolocation API](/en-US/docs/Web/API/Geolocation_API)." in a note below the top intro.

When the user activates the control, they are presented with a dialog box that asks them for permission to share their location data. If they grant permission, the browser will attempt to retrieve the user's location data using the Geolocation API in the background.

> By default, the browser requests location data once, as if the {{domxref("Geolocation.getCurrentPosition()")}} method was called. However, if the [`watch`](/en-US/docs/Web/HTML/Reference/Elements/geolocation#watch) attribute is set to `true`, the browser constantly requests the data, as if {{domxref("Geolocation.watchPosition()")}} was called.

Reviewer: Constantly requests the data makes me wonder how often etc. Perhaps better to do like this

```suggestion
By default, the browser requests location data once, as if the {{domxref("Geolocation.getCurrentPosition()")}} method was called. However, if the [`watch`](/en-US/docs/Web/HTML/Reference/Elements/geolocation#watch) attribute is set to `true`, the browser updates the element whenever the device moves, as if {{domxref("Geolocation.watchPosition()")}} was called.
```
